contrib/bsddialog/lib/barbox.c

```diff
@@ -249,7 +267,21 @@ do_mixedgauge(struct bsddialog_conf *conf, const char *text, int rows, int cols,
 		/* label */
 		if (color && miniperc >= 0)
 			wattron(d.widget, A_BOLD);
-		mvwaddstr(d.widget, i+1, 2, CHECK_STR(minilabels[i]));
+		if (label_len_cap > 0 &&
+		    (int)strcols(CHECK_STR(minilabels[i])) > label_len_cap) {
+			char *trunclabel, *seeklabel;
+
+			trunclabel = strdup(minilabels[i]);
```

Comment:
I guess this should be CHECK_STR now given it wants to treat NULL as "" (questionable, but...)

Reply:
Although I guess one can also argue that it's impossible to hit since "" should have strcols return 0, and label_len_cap is never negative...

Reply:
Yeah, that was my reasoning for not using CHECK_STR here.

Reply:
(Also, I found two bugs in this change so have updated it, will re-push in a bit)